### DISCUSSION TITLE
Use first seed when batching images instead of last seed.

### DISCUSSION THREAD

**Discussion by Gecktendo** [Kilvoctu/aiyabot#246] [General]
When I run aiyabot in batches larger than 1, the last seed is grabbed when I make edits to the prompt instead of the first seed. The result is that I'm not editing the same x seeds, I have to subtract x number of images to get back to the first seed in the image.

**Comment by wizz13150** [comment]
Hey, 
for this you can just comment the line 515 in stablecog.py. It will do the job and keep the first seed.

https://github.com/Kilvoctu/aiyabot/blob/8ee436d05b253f906416073d0c7bc52d0d581dd1/core/stablecog.py#L515

![image](https://github.com/Kilvoctu/aiyabot/assets/22177081/7db46286-6cf4-4ac7-845d-45bf34a0a12e)

![image](https://github.com/Kilvoctu/aiyabot/assets/22177081/1d487156-0949-4fc0-a021-23ca6a9fc7fe)


Cheers ! 🥂

**Reply by Gecktendo** [reply]
Wait, so that line was intentional? What purpose did it serve? I thought it was a bug.

**Reply by wizz13150** [reply]
Same lol. 
Not sure ((:

**Reply by Gecktendo** [reply]
I have another little bug. When I set seed to 0 and I try to pull up the edit, it doesn't preserve the 0 and thus ends up randomizing the seed. I get that this is a silly edge case but I think it should return 0, not "".

**Reply by wizz13150** [reply]
Hey, @Gecktendo, sorry for the delay !

Hum, indeed, this is a sneaky one !

I won't mention that using 0 as seed is probably not part of best practice with SD lol, you obviously knows it.
My fix for this would be to simply disallow 0 as seed, ahah.

jk jk

Here is what you're looking for, this will allow 0 as value.
Just edit line 77 in `viewhandler.py` : 

https://github.com/Kilvoctu/aiyabot/blob/05840c2b00328c67026737667978a91237ba9cf5/core/viewhandler.py#L77
to
`value=str(input_tuple[10]),`
<br>

This will do the job.

Screen :
![image](https://github.com/Kilvoctu/aiyabot/assets/22177081/7fdb3fae-3cd5-41c4-a5aa-7466332d3244)



Cheers ! 🥂

**Comment by Kilvoctu** [comment]
> Wait, so that line was intentional? What purpose did it serve? I thought it was a bug.

Looks like leftover code which no longer behaved properly ever since PR #170 was merged. I'd call it a bug.

Thanks for the fix @wizz13150 . I really don't like touching the batch stuff code. If it was up to me, I'd just remove the feature altogether; bad past experiences.
